Добавление Скачивание и Удаление аватара пользоавтеля

vzeten opened the pull request:
build.gradle.kts: - добавил парочку зависимостей
default-user-avatar.png: Добавил картинку-заглушку для аватара по умолчанию. org.mockito.plugins.MockMaker - добавил чтобы тест один запускался PictureDto.java - ДТО для аватарок
PictureType.java - enum для типов аватаров
S3Config.java - подключение к AWS
S3service.java - работа с S3
User.java - Добавил @Builder.Default к mentee и mentor UserDto.java - Добавил поле List<PictureDto> pictures для аватров UserMapper.java - Добавил маппинг для аватарок (PictureDto) AvatarController.java - Эндпоинты по задаче
AvatarServiceImpl.java - реализация методов к задаче

Commit:
build.gradle.kts: - добавил парочку зависимостей
default-user-avatar.png: Добавил картинку-заглушку для аватара по умолчанию.
org.mockito.plugins.MockMaker - добавил чтобы тест один запускался
PictureDto.java - ДТО для аватарок
PictureType.java - enum для типов аватаров
S3Config.java - подключение к AWS
S3service.java - работа с S3
User.java - Добавил @Builder.Default к mentee и mentor
UserDto.java - Добавил поле List<PictureDto> pictures для аватров
UserMapper.java - Добавил маппинг для аватарок (PictureDto)
AvatarController.java - Эндпоинты по задаче
AvatarServiceImpl.java - реализация методов к задаче

Commit:
поправил Checkstyle

7glebalyot commented on src/main/java/school/faang/user_service/controller/avatar/AvatarController.java:
```diff
@@ -0,0 +1,101 @@
+    @Parameter(name = "x-user-id", required = true, in = ParameterIn.HEADER)
+    public ResponseEntity<List<PictureDto>> uploadAvatar(@RequestParam("file") MultipartFile file) {
+        long userId = userContext.getUserId();
+        log.info("Received request to upload avatar for user ID: {}. File name: {}, size: {} bytes",
```
мне кажется что в логе, будет лишнее добавлять прям имя файла и его размеры. Как будто избыточные расчеты будут проводиться просто для логирования. Но это просто мнение, не знаю как правильно, можно посоветоваться с Лешей

vzeten replied:
Да могу и убрать не проблема, хотя вес в байтайх наверное можно оставить

7glebalyot commented on src/main/java/school/faang/user_service/entity/user/User.java:
```diff
@@ -103,12 +103,15 @@ public class User {
     private List<User> mentees = new ArrayList<>();
 
     @ManyToMany
+
```
тут возможно лишний пробел

vzeten replied:
Да точно) спасибо!

vzeten commented on src/main/resources/static/images/default-user-avatar.png:
Фотка огонь!

Commit:
добавил @Retryable на методы работающие с AWS
исправил проверку на дефолтную картинку с http на параметр dicebearBaseUrl
убрал пробел в User

Commit:
Вернул проверку на дефолтную картинку по "http" а не по dicebearBaseUrl, иначе тест ломался

AlexeyAntipin commented on src/main/java/school/faang/user_service/config/S3Config.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    @Bean("bucketName")
+    public String bucketName() {
```
Нет смысла делать это бином, лучше ограничиваться обычным @Value

AlexeyAntipin commented on src/main/java/school/faang/user_service/service/avatar/AvatarServiceImpl.java:
```diff
@@ -0,0 +1,191 @@
+        }
+
+        try {
+            BufferedImage originalImage = ImageIO.read(file.getInputStream());
```
Эту логику лучше вынести в какой-нибудь специализированные сервис для s3. Что-то типа S3Service

AlexeyAntipin commented on src/main/java/school/faang/user_service/service/avatar/AvatarServiceImpl.java:
```diff
@@ -0,0 +1,191 @@
+     *
+     * @param user пользователь, чей аватар нужно проверить и удалить.
+     */
+    @Retryable(retryFor = {
```
У тебя сейчас Retryable достаточно бесполезный, потому что внутри ты все оборачиваешь в try catch и ошибка не будет проброшена

AlexeyAntipin commented on src/main/java/school/faang/user_service/service/s3/S3service.java:
```diff
@@ -0,0 +1,105 @@
+        return s3Client.getUrl(bucketName, fileId).toString();
+    }
+
+    @Retryable(retryFor = {
```
То же самое с try catch, либо нужно не оборачивать в try catch, либо убирать retryable

Commit:
AvatarServiceImpl:
Теперь отвечает только за свою бизнес-логику: проверить файл, найти пользователя, вызвать S3service и сохранить результат.
Вся работа с картинками (изменение размера) из него убрана.

S3service:
Появился новый удобный метод uploadAvatar, который инкапсулирует всю логику работы с аватарами.
Старый метод uploadFileToS3 остался без изменений для поддержки обратной совместимости.
Исправлена работа @Retryable (убраны try-catch и лишние исключения).
Имя бакета теперь внедряется напрямую через @Value, что является лучшей практикой.

S3Config:
Убран лишний бин для bucketName. Конфигурация стала чище.

AvatarServiceImplTest:
Тесты были упрощены и теперь проверяют только то, за что отвечает AvatarServiceImpl, не залезая во внутреннюю реализацию S3service.
